reconcile/utils/oc.py

```diff
@@ -1016,16 +1134,24 @@ def __init__(
             init_api_resources=False,
             local=local,
             insecure_skip_tls_verify=insecure_skip_tls_verify,
+            connection_parameters=connection_parameters,
         )
 
+        if connection_parameters:
+            token = connection_parameters.automation_token
+            if connection_parameters.is_cluster_admin:
+                token = connection_parameters.cluster_admin_automation_token
+
+            server = connection_parameters.server_url
+
         if server:
             self.client = self._get_client(server, token)
             self.api_kind_version = self.get_api_resources()
             self.api_resources = self.api_kind_version.keys()
         else:
             raise Exception("A method relies on client/api_kind_version to be set")
 
-        self.object_clients = {}
+        self.object_clients: dict[Any, Any] = {}
```

Comment:
Shouldn't this be a `Mapping[Any, Any]` or Mapping[str, Any]

Reply:
I think in this case it is ok because we instantiate a concrete object here. `Mapping`, `Iterable` and such are used in function arguments to keep them more flexible for consumers. (https://service.pages.redhat.com/dev-guidelines/docs/appsre/python-style-guide/#abstract-collection-types)